plugins/Compressor/CompressorControls.h

```diff
@@ -0,0 +1,101 @@
+/*
+ * CompressorControls.h
+ *
+ * Copyright (c) 2020 Lost Robot <r94231@gmail.com>
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#ifndef COMPRESSOR_CONTROLS_H
+#define COMPRESSOR_CONTROLS_H
+
+#include "EffectControls.h"
+#include "CompressorControlDialog.h"
+#include "Knob.h"
+
+
+class CompressorEffect;
+
+
+class CompressorControls : public EffectControls
+{
+	Q_OBJECT
+public:
+	CompressorControls(CompressorEffect* effect);
+
+	void saveSettings(QDomDocument & _doc, QDomElement & _parent) override;
+	void loadSettings(const QDomElement & _this) override;
+	inline QString nodeName() const override
+	{
+		return "CompressorControls";
+	}
+
+	int controlCount() override
+	{
+		return 31;
```

Comment:
Not sure what was going through my head back then.  Maybe I was counting GUI elements instead of models for some reason.  I'm still not sure what that function is used for, I'll check.

Reply:
For now, it doesn't matter unless you put 0 here. LMMS uses this value only to check if a plugin has any controls.